src/object/obj_enum.c

```diff
@@ -273,15 +273,15 @@ fill_rec(daos_handle_t ih, vos_iter_entry_t *key_ent, struct dss_enum_arg *arg,
 	 * If we've decided to inline the data, append the data to iovs.
 	 * NB: Punched recxs do not have any data to copy.
 	 */
-	if (inline_data && data_size > 0) {
+	if (inline_data && data_size > 0 && arg->copy_cb) {
 		d_iov_t iov_out;
 
 		/* inline packing for the small recx located on SCM */
 		D_ASSERT(key_ent->ie_biov.bi_addr.ba_type == DAOS_MEDIA_SCM);
 
 		d_iov_set(&iov_out, iovs[arg->sgl_idx].iov_buf +
 				       iovs[arg->sgl_idx].iov_len, data_size);
-		rc = vos_iter_copy(ih, key_ent, &iov_out);
+		rc = arg->copy_cb(ih, key_ent, &iov_out);
```

Comment:
why it's changed to use a callback? I think VOS interface can be called by object module.

Reply:
There will be compile error if we call vos_iter_copy() from the obj_enum.c because of the obj_enum.c will be shared by both client and server.

Reply:
I suppose enum packing code should be server side only, right?

Reply:
The pack is server side logic, but pack and unpack shares some functions, so they are in implemented the same source file.